[Executorch][XNNPACK] Fix broken xnnpack build

kimishpatel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #1229

Summary:
Recent change to edge dialect verifier enables checking if the graph
contains only aten ops. This broke XNNPACK delegate because linear
transform converts addmm and friends to linear.

This fix just works around it.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D51406090](https://our.internmc.facebook.com/intern/diff/D51406090)

kimishpatel committed:
[Executorch][XNNPACK] Fix broken xnnpack build

Summary:
Recent change to edge dialect verifier enables checking if the graph
contains only aten ops. This broke XNNPACK delegate because linear
transform converts addmm and friends to linear.

This fix just works around it.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51406090).

mcr229 commented on backends/xnnpack/xnnpack_preprocess.py:
```diff
@@ -37,7 +37,9 @@
 )
+from executorch.exir.verification.verifier import EXIREdgeDialectVerifier
 from torch._export.exported_program import ExportedProgram
+from torch._export.verifier import Verifier
```
I think this is unused

mcr229 commented on backends/xnnpack/xnnpack_preprocess.py:
```diff
@@ -189,6 +190,26 @@ def preprocess(
+            ep.equality_constraints,
+            copy.deepcopy(ep.module_call_graph),
+            ep.example_inputs,
+            verifier=EXIREdgeDialectVerifier(check_edge_ops=False, enable=False, class_only=True),
```
lint issue on this line

kimishpatel committed:
Update on "[Executorch][XNNPACK] Fix broken xnnpack build"


Summary:
Recent change to edge dialect verifier enables checking if the graph
contains only aten ops. This broke XNNPACK delegate because linear
transform converts addmm and friends to linear.

This fix just works around it.

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D51406090](https://our.internmc.facebook.com/intern/diff/D51406090)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51406090).
